### ISSUE TITLE
pipeline file contains reference to potentially invalid custom component image

### ISSUE THREAD

**Issue by ptitzler** [elyra-ai/elyra#2773]
**Describe the issue**
For custom components the .pipeline file contains references to a deployment specific image URL (note the port, which is not the Jupyter default): 

```
           "ui_data": {
              "label": "Split File",
              "image": "http://localhost:8890/static/elyra/kubeflow.svg",
              "x_pos": 219,
              "y_pos": 283.5,
              "description": "Split file into two"
            }
```

While the image appears to be displayed properly in the VPE irrespective of whether the specified port is the port where JupyterLab was started, the metadata should probably not include protocol, host and port information.

**To Reproduce**
1. Launch JupyterLab specifying a non-default port (`--port xxxx`)
2. Create a KFP/AA pipeline and add a custom component
3. Save the pipeline and review the .pipeline file

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 3.9.0

**Comment by ajbozarth** [comment]
That value is actually a "guaranteed" value, meaning that on open/edit/validate it checks it's value against the one in the current palette and if it doesn't match it replaces it with the current value. This means that in practice this persisted value is always updated prior to being used by the UI. Also since it is contained in the `ui_data` black box, it's value only matters to the UI, which handles this properly.

tl;dr despite it "looking" bad when reading the json there is no actual problem here since the value is always validated and updated prior to use

**Comment by ajbozarth** [comment]
After some offline discussion we decided we still dont want to persist the URL for PI Security reasons so I've opened a PR to deal with this.
